packages/console/src/pages/SignInExperience/utilities.ts

```diff
@@ -98,3 +100,51 @@ export const compareSignInMethods = (
 
   return Object.values(SignInMethodKey).every((key) => beforeMethods[key] === afterMethods[key]);
 };
+
+export const flattenObject = (
+  object: Record<string, unknown>,
+  keyPrefix = ''
+): Record<string, unknown> => {
+  return Object.keys(object).reduce((result, key) => {
+    const prefix = keyPrefix ? `${keyPrefix}.` : keyPrefix;
+    const dataKey = `${prefix}${key}`;
+    const data = object[key];
+
+    const temporaryObject =
+      typeof data === 'object'
+        ? // eslint-disable-next-line no-restricted-syntax
+          flattenObject(data as Record<string, unknown>, dataKey)
+        : { [dataKey]: data };
```

Comment:
I think it's better to use zod to guard the data type

Reply:
Add an issue to track this: https://linear.app/silverhand/issue/LOG-4235/handle-type-assertion-gracefully-in-sie-utilities

Reply:
updated with type assertion